src/webgl/p5.RendererGL.Retained.js

```diff
@@ -30,11 +28,9 @@ p5.RendererGL.prototype._initBufferDefaults = function(gId) {
   this._freeBuffers(gId);
 
   //@TODO remove this limit on hashes in retainedMode.geometry
-  hashCount++;
-  if (hashCount > 1000) {
+  if (Object.keys(this.retainedMode.geometry).length > 1000) {
```

Comment:
Why do we impose this 1000 hash limit? Is that just an arbitrary number?

Reply:
I think so, kind of an arbitrary limit intended to make sure we don't run out of GPU memory. I think we could maybe be a bit smarter about this though -- I think `gl.bufferData()` throws `gl.OUT_OF_MEMORY` if we do run out?